lib/compilers/v.ts

```diff
@@ -92,10 +92,31 @@ export class VCompiler extends BaseCompiler {
         return path.join(dirPath, 'output' + this.outputFileExt);
     }
 
-    override getDefaultExecOptions(): ExecutionOptions & {env: Record<string, string>} {
-        const options = super.getDefaultExecOptions();
-        options.env['VMODULES'] = path.join(path.dirname(this.compiler.exe), '.vmodules');
-        return options;
+    override async runCompiler(
+        compiler: string,
+        options: string[],
+        inputFilename: string,
+        execOptions: ExecutionOptions & {env: Record<string, string>},
+    ): Promise<CompilationResult> {
+        if (!execOptions) {
+            execOptions = super.getDefaultExecOptions();
+        }
+        execOptions.env['VMODULES'] = path.join(this.getOutputDirFromOptions(options), '.vmodules');
```

Comment:
Sorry for leading you astray: I would suggest using input filename's directory! That's guaranteed to be a temporary directory and is actually also where the output will go. and it's less code :)

Reply:
No problem